### PR TITLE
[Caching] Add test fixtures for element state testing

### PR DESCRIPTION
**Pull request by kyleve**

### CODE REVIEW

File: BlueprintUI/Sources/Element/ElementState.swift [hunk 1]
```diff
@@ -542,6 +632,68 @@ extension Dictionary {
...
+
+    func treeDidPerformCachedLayout(
+        _ layout: ElementState.CachedLayout,
+        for state: ElementState
+    )
+}
+
+
+extension Optional where Wrapped == ElementStateTreeDelegate {
+    func perform(_ block: (Wrapped) -> Void) {
```
Comment (n8chur) [inline BlueprintUI/Sources/Element/ElementState.swift:688]: Could we name this function in a way where it's clear that it's only performed in `DEBUG` mode? `ifDebugPerform`?

**Review by n8chur — COMMENTED** [review]
Should this PR be targetting `kareem/elementidentifier-changes` as the base branch?

File: BlueprintUI/Sources/Element/ElementState.swift (hunk 1 above)
Comment (kyleve) [reply BlueprintUI/Sources/Element/ElementState.swift:705]: How about just `ifDebug`?

Comment (kyleve) [reply BlueprintUI/Sources/Element/ElementState.swift:704]: This is done in a future branch

**Review by kdaker — APPROVED** [review]

**Pull request merged**